### PR TITLE
[WIP] Renaming

### PR DESCRIPTION
**Pull request by josh146**
**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/operation.py [hunk 1]
```diff
@@ -561,7 +561,33 @@ def compute_matrix(*params, **hyperparams):  # pylint:disable=unused-argument
...
-    def matrix(self, wire_order=None):
+    @property
+    def matrix(self):
+        r"""Matrix representation of an instantiated operator
+        in the computational basis.
+
+        .. warning::
+
+            The ``matrix`` property is deprecated and will be removed in
+            PennyLane version 0.23.
```
Comment (mariaschuld) [inline pennylane/operation.py:572]: Should we be less specific here? This may not be a priority to remove so fast?

File: pennylane/operation.py [hunk 2]
```diff
@@ -1225,7 +1277,20 @@ def inv(self):
...
         return self
 
-    def matrix(self, wire_order=None):
+    @property
+    def matrix(self):
+        warnings.warn(
+            "The 'matrix' property is deprecated, and will be replaced with a method 'matrix()' in v0.23",
+            UserWarning,
+        )
+        op_matrix = self._matrix(*self.parameters)
```
Comment (mariaschuld) [inline pennylane/operation.py:1286]: I'm confused...do we still HAVE `_matrix`? :) Shouldn't this call `get_matrix`?

**Review by mariaschuld — CHANGES_REQUESTED** [review]
Wonderful, thanks Josh!

Just the one question about calling `_matrix`.

And maybe add a test for the deprecation warning? The tests should not call `matrix` as a property anywhere (and I am surprised that codecov does not complain?)

File: pennylane/operation.py (hunk 1 above)
Comment (josh146) [reply pennylane/operation.py:572]: I think this is a good idea!

File: pennylane/operation.py (hunk 2 above)
Comment (josh146) [reply pennylane/operation.py:1286]:
```suggestion
        op_matrix = self.get_matrix()
```

oops, thanks for catching this one

**Comment by josh146** [comment]
@mariaschuld thanks for the good catches, have now fixed all failing tests!

**Review by mariaschuld — APPROVED** [review]

**Pull request merged**